### PR TITLE
fixing IDENTITY-3413 Split AUTHZ_USER column of IDN_OAUTH2_ACCESS_TOKEN table into USER_DOMAIN and AUTHZ_USER

### PR DESCRIPTION
**Pull request by madurangasiriwardena**

### CODE REVIEW

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java [hunk 1]
```diff
@@ -173,7 +174,7 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                 }
                 return false;
             }
-            tokReqMsgCtx.setAuthorizedUser(resourceOwnerUserName);
+            tokReqMsgCtx.setAuthorizedUser(OAuth2Util.getUserFromUserName(resourceOwnerUserName));
         } else {
             log.debug("Cannot find a Subject in the Assertion");
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java]: Check ```isDebugEnabled()```

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java [hunk 2]
```diff
@@ -48,8 +48,7 @@ public void run() {
                     if (authContextTokenDO.getAuthzCodeDO() == null) {
                         log.debug("Auth Token Data removing Task is started to run");
                         TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
-                        tokenMgtDAO.deactivateAuthorizationCode(authContextTokenDO.getAuthzCode(), authContextTokenDO
-                                .getTokenId());
+                        tokenMgtDAO.doExpireAuthzCode(authContextTokenDO.getAuthzCode());
                     } else {
                         log.debug("Auth Token Data persisting Task is started to run");
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java]: Check ```isDebugEnabled()```

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java [hunk 3]
```diff
@@ -48,8 +48,7 @@ public void run() {
                     if (authContextTokenDO.getAuthzCodeDO() == null) {
                         log.debug("Auth Token Data removing Task is started to run");
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java]: Check ```isDebugEnabled()```

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java (hunk 1 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java]: fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java (hunk 2 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java]: fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java (hunk 3 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthPersistenceTask.java]: fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 4]
```diff
@@ -513,17 +551,24 @@ public AuthzCodeDO validateAuthorizationCode(String consumerKey, String authoriz
...
-                    return new AuthzCodeDO(authorizedUser,
-                                           OAuth2Util.buildScopeArray(scopeString),
-                                           issuedTime, validityPeriod, callbackUrl);
+                    String userstoreDomain = resultSet.getString(2);
+                    int tenantId = resultSet.getInt(3);
+                    String tenantDomain = OAuth2Util.getTenantDomain(tenantId);
+                    String scopeString = resultSet.getString(4);
+                    String callbackUrl = resultSet.getString(5);
+                    Timestamp issuedTime = resultSet.getTimestamp(6,
+                            Calendar.getInstance(TimeZone.getTimeZone("UTC")));
```
Comment (pulasthi7) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Use a constant for "UTC"

Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java [hunk 5]
```diff
@@ -123,4 +123,13 @@ public String getUserName() {
     public void setUserName(String userName) {
         this.userName = userName;
     }
+
+    @Override
+    public String toString(){
```
Comment (omindu) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 6]
```diff
@@ -853,36 +913,77 @@ public void setAccessTokenState(Connection connection, String tokenId, String to
...
+            String sqlQuery = SQLQueries.REVOKE_ACCESS_TOKEN.replace(IDN_OAUTH2_ACCESS_TOKEN, accessTokenStoreTable);
             connection = IdentityDatabaseUtil.getDBConnection();
             ps = connection.prepareStatement(sqlQuery);
-            ps.setString(1, OAuthConstants.TokenStates.TOKEN_STATE_REVOKED);
-            ps.setString(2, UUID.randomUUID().toString());
-            ps.setString(3, persistenceProcessor.getProcessedAccessTokenIdentifier(token));
-            int count = ps.executeUpdate();
-            if (log.isDebugEnabled()) {
-                log.debug("Number of rows being updated : " + count);
+            for (String token: tokens){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 7]
```diff
@@ -1107,6 +1215,294 @@ public void revokeOAuthConsentByApplicationAndUser(String username, String appli
...
+        ResultSet rs = null;
+
+        List<AuthzCodeDO> latestAuthzCodes = new ArrayList<>();
+        try {
+            String sqlQuery = SQLQueries.LIST_LATEST_AUTHZ_CODES_IN_TENANT;
+            connection = IdentityDatabaseUtil.getDBConnection();
+            ps = connection.prepareStatement(sqlQuery);
+            ps.setInt(1, tenantId);
+            rs = ps.executeQuery();
+            while (rs.next()){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 8]
```diff
@@ -1107,6 +1215,294 @@ public void revokeOAuthConsentByApplicationAndUser(String username, String appli
...
+
+        List<AuthzCodeDO> latestAuthzCodes = new ArrayList<>();
+        try {
+            String sqlQuery = SQLQueries.LIST_LATEST_AUTHZ_CODES_IN_USER_DOMAIN;
+            connection = IdentityDatabaseUtil.getDBConnection();
+            ps = connection.prepareStatement(sqlQuery);
+            ps.setInt(1, tenantId);
+            ps.setString(2, userStorDomain.toUpperCase());
+            rs = ps.executeQuery();
+            while (rs.next()){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+import org.wso2.carbon.identity.oauth.OAuthUtil;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dao.TokenMgtDAO;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+import org.wso2.carbon.identity.oauth2.model.AuthzCodeDO;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.identity.user.store.configuration.listener.UserStoreConfigListener;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.util.*;
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Better if you can remove * imports.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                OAuthUtil.clearOAuthCache(accessTokenDO.getAccessToken());
+            }
+            ArrayList<String> tokensToRevoke = new ArrayList<>();
+            for (Map.Entry entry : latestAccessTokens.entrySet()){
+                tokensToRevoke.add(((AccessTokenDO) entry.getValue()).getAccessToken());
+            }
+            tokenMgtDAO.revokeTokens(tokensToRevoke.toArray(new String[tokensToRevoke.size()]));
+            List<AuthzCodeDO> latestAuthzCodes = tokenMgtDAO.getLatestAuthorizationCodesOfUserStore(tenantId,
+                    userStoreName);
+            for (AuthzCodeDO authzCodeDO : latestAuthzCodes){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/CarbonOAuthTokenRequest.java [hunk 11]
```diff
@@ -60,6 +60,9 @@ public CarbonOAuthTokenRequest(HttpServletRequest request) throws OAuthSystemExc
         assertion = request.getParameter(ASSERTION);
         credentialType = request.getParameter(CREDENTIAL_TYPE);
         tenantDomain = request.getParameter(TENANT_DOMAIN);
+        if (tenantDomain == null){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/CarbonOAuthTokenRequest.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 12]
```diff
@@ -568,4 +570,20 @@ private static String getDomainFromName(String name) {
         return UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
     }
 
+    public static User getUserFromUserName(String username) throws IllegalArgumentException{
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dao.TokenMgtDAO;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+import org.wso2.carbon.identity.oauth2.model.AuthzCodeDO;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.stratos.common.beans.TenantInfoBean;
+import org.wso2.carbon.stratos.common.exception.StratosException;
+import org.wso2.carbon.stratos.common.listeners.TenantMgtListener;
+
+import java.util.*;
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Better if you can avoid * imports.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+        TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
+        try {
+            Set<AccessTokenDO> accessTokenDOs = tokenMgtDAO.getAccessTokensOfTenant(tenantId);
+            Map<String, AccessTokenDO> latestAccessTokens = new HashMap<>();
+            for (AccessTokenDO accessTokenDO : accessTokenDOs) {
+                String keyString = accessTokenDO.getConsumerKey() + ":" + accessTokenDO.getAuthzUser() + ":" +
+                        OAuth2Util.buildScopeString(accessTokenDO.getScope());
+                AccessTokenDO accessTokenDOFromMap = latestAccessTokens.get(keyString);
+                if (accessTokenDOFromMap != null) {
+                    if (accessTokenDOFromMap.getIssuedTime().before(accessTokenDO.getIssuedTime())){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+                }
+
+                //Clear cache
+                OAuthUtil.clearOAuthCache(accessTokenDO.getConsumerKey(), accessTokenDO.getAuthzUser(),
+                        OAuth2Util.buildScopeString(accessTokenDO.getScope()));
+                OAuthUtil.clearOAuthCache(accessTokenDO.getConsumerKey(), accessTokenDO.getAuthzUser());
+                OAuthUtil.clearOAuthCache(accessTokenDO.getAccessToken());
+            }
+            ArrayList<String> tokensToRevoke = new ArrayList<>();
+            for (Map.Entry entry : latestAccessTokens.entrySet()){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+                tokensToRevoke.add(((AccessTokenDO) entry.getValue()).getAccessToken());
+            }
+            tokenMgtDAO.revokeTokens(tokensToRevoke.toArray(new String[tokensToRevoke.size()]));
+            List<AuthzCodeDO> latestAuthzCodes = tokenMgtDAO.getLatestAuthorizationCodesOfTenant(tenantId);
+            for (AuthzCodeDO authzCodeDO : latestAuthzCodes){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+        TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
+        try {
+            Set<AccessTokenDO> accessTokenDOs = tokenMgtDAO.getAccessTokensOfUserStore(tenantId, userStoreName);
+            Map<String, AccessTokenDO> latestAccessTokens = new HashMap<>();
+            for (AccessTokenDO accessTokenDO : accessTokenDOs) {
+                String keyString = accessTokenDO.getConsumerKey() + ":" + accessTokenDO.getAuthzUser() + ":" +
+                        OAuth2Util.buildScopeString(accessTokenDO.getScope());
+                AccessTokenDO accessTokenDOFromMap = latestAccessTokens.get(keyString);
+                if (accessTokenDOFromMap != null) {
+                    if (accessTokenDOFromMap.getIssuedTime().before(accessTokenDO.getIssuedTime())){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+                }
+
+                //Clear cache
+                OAuthUtil.clearOAuthCache(accessTokenDO.getConsumerKey(), accessTokenDO.getAuthzUser(),
+                        OAuth2Util.buildScopeString(accessTokenDO.getScope()));
+                OAuthUtil.clearOAuthCache(accessTokenDO.getConsumerKey(), accessTokenDO.getAuthzUser());
+                OAuthUtil.clearOAuthCache(accessTokenDO.getAccessToken());
+            }
+            ArrayList<String> tokensToRevoke = new ArrayList<>();
+            for (Map.Entry entry : latestAccessTokens.entrySet()){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java (hunk 9 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Fixed with 43f6a5676e8e49c0d9595d500d056a836e0b5177

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java (hunk 13 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Fixed with 43f6a5676e8e49c0d9595d500d056a836e0b5177

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java (hunk 5 above)
Comment (madurangasiriwardena) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 6 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 7 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 8 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java (hunk 10 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/CarbonOAuthTokenRequest.java (hunk 11 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/CarbonOAuthTokenRequest.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java (hunk 12 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java (hunk 14 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java (hunk 15 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java (hunk 16 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthTenantMgtListenerImpl.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java (hunk 17 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java (hunk 18 above)
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthUserStoreConfigListenerImpl.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1679

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java (hunk 5 above)
Comment (dinusha-dilrukshi) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/User.java]: Add method level comments. Method name seems to misleading.

**Pull request merged**